framework/src/variables/MooseVariableBase.C

```diff
@@ -105,7 +105,8 @@ MooseVariableBase::MooseVariableBase(const InputParameters & parameters)
     _tid(getParam<THREAD_ID>("tid")),
     _count(getParam<unsigned int>("components")),
     _use_dual(getParam<bool>("use_dual")),
-    _is_array(getParam<bool>("array"))
+    _is_array(getParam<bool>("array")),
+    _is_lower_d(blocksMaxDimension() < _mesh.dimension())
```

Comment:
I m not sure it s that clear cut.

if we have a mesh that has both 2D and 3D regions, completely split not 2D living on the 3D sideset, then a 2D variable would not be lower D necessarily

Reply:
ok I will push an additional commit to see if you like it better